lib/printers/default.js

```diff
@@ -1,15 +1,52 @@
 'use strict';
 
+const path = require('path');
+
 class DefaultPrinter {
   constructor(options = {}) {
     this.delegates = [];
 
     if (options.json) {
-      let JsonPrinter = require('./json');
-      this.delegates.push(new JsonPrinter(options));
-    } else {
-      let PrettyPrinter = require('./pretty');
-      this.delegates.push(new PrettyPrinter(options));
+      options.printer = 'json';
+    }
+
+    switch (options.printer) {
+      case 'json': {
+        let JsonPrinter = require('./json');
+        this.delegates.push(new JsonPrinter(options));
+        break;
+      }
+      case 'pretty': {
+        let PrettyPrinter = require('./pretty');
+        this.delegates.push(new PrettyPrinter(options));
+        break;
+      }
+      default: {
+        const { dir } = path.parse(options.printer);
+
+        // printers can be relative to the current working directory is come from an npm package
+        if (dir === '.') {
+          const customPrinterPath = path.resolve(process.cwd(), options.printer);
```

Comment:
We shouldn't use the `process.cwd()` here (we've landed a threaded through working directory properly)

Reply:
Also I think I'd prefer to have the bin script resolve these things itself, and pass it in

Reply:
will make the changes to do the resolution in the bin script.